library/blockly/html/generator.js

```diff
@@ -563,6 +563,23 @@ htmlGen['summary'] = function(block){
     return code;
 };
 
+htmlGen['rgba_picker'] = function(block){
+	var r = looseEscape(block.getFieldValue('r'));
+	var g = looseEscape(block.getFieldValue('g'));
+	var b = looseEscape(block.getFieldValue('b'));
+	var a = looseEscape(block.getFieldValue('a'));
+	
+	return "rgba(" + r + ", " + g + ", " + b + ", " + a + ")";
+}
+
+htmlGen['hex_picker'] = function(block){
+	return "#" + looseEscape(block.getFieldValue('color'));
```

Comment:
We should be stricter here - it needs a sort of 'hexadecimal escape' to ensure only hexadecimal digits can be entered.